src/cmd/flash.js

```diff
@@ -55,29 +55,46 @@ module.exports = class FlashCommand extends CLICommandBase {
 	}
 
 	async flashOverUsb({ binary, factory }) {
-		if (utilities.getFilenameExt(binary) === '.zip') {
-			throw new Error("Use 'particle flash --local' to flash a zipped bundle.");
-		}
+		let device;
+		try {
+			if (utilities.getFilenameExt(binary) === '.zip') {
+				throw new Error("Use 'particle flash --local' to flash a zipped bundle.");
+			}
 
-		const device = await usbUtils.getOneUsbDevice({ ui: this.ui });
-		const platformName = platformForId(device.platformId).name;
-		validateDFUSupport({ device, ui: this.ui });
+			device = await usbUtils.getOneUsbDevice({ ui: this.ui });
+			const platformName = platformForId(device.platformId).name;
+			validateDFUSupport({ device, ui: this.ui });
 
-		let files;
-		const knownAppPath = knownAppsForPlatform(platformName)[binary];
-		if (knownAppPath) {
-			files = [knownAppPath];
-		} else {
-			files = [binary];
-		}
-		const modulesToFlash = await parseModulesToFlash({ files });
+			let files;
+			const knownAppPath = knownAppsForPlatform(platformName)[binary];
+			if (knownAppPath) {
+				files = [knownAppPath];
+			} else {
+				files = [binary];
+			}
 
-		await this._validateModulesForPlatform({ modules: modulesToFlash, platformId: device.platformId, platformName });
-		const flashSteps = await createFlashSteps({ modules: modulesToFlash, isInDfuMode: device.isInDfuMode , platformId: device.platformId, factory });
+			const modulesToFlash = await parseModulesToFlash({ files });
+
+			await this._validateModulesForPlatform({
+				modules: modulesToFlash,
+				platformId: device.platformId,
+				platformName
+			});
+			const flashSteps = await createFlashSteps({
+				modules: modulesToFlash,
+				isInDfuMode: device.isInDfuMode,
+				platformId: device.platformId,
+				factory
+			});
 
-		this.ui.write(`Flashing ${platformName} device ${device.id}`);
-		const resetAfterFlash = !factory && modulesToFlash[0].prefixInfo.moduleFunction === ModuleInfo.FunctionType.USER_PART;
-		await flashFiles({ device, flashSteps, resetAfterFlash, ui: this.ui });
+			this.ui.write(`Flashing ${platformName} device ${device.id}`);
+			const resetAfterFlash = !factory && modulesToFlash[0].prefixInfo.moduleFunction === ModuleInfo.FunctionType.USER_PART;
+			await flashFiles({ device, flashSteps, resetAfterFlash, ui: this.ui });
+		} catch (error) {
+			if (device) {
+				await device.close();
+			}
+		}
```

Comment:
@hugomontero and I figured that device doesn't close and it leaves the terminal hanging if the logic fails in any of the above lines such as `const modulesToFlash = await parseModulesToFlash({ files });` . We are if we should add this try/catch block here in this file for other methods (which is `flashLocal`) or if there is a better way to handle it.